### PR TITLE
ICP-8841 Fans sometimes return values between 1 and 33 for low

### PR DESCRIPTION
**Pull request by greens**

### CODE REVIEW

File: devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy [hunk 1]
```diff
@@ -110,7 +110,9 @@ def fanEvents(physicalgraph.zwave.Command cmd) {
 	def value = (cmd.value ? "on" : "off")
 	def result = [createEvent(name: "switch", value: value)]
 	result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value)
-	result << createEvent(name: "fanSpeed", value: Math.round(cmd.value/33))
+	def fan_level = Math.round(cmd.value/33)
+	if (cmd.value < 33 * cmd.value > 1) fan_level = 1 //sometimes we get "1" when the device is on low
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: Isn't `cmd.value < 33 * cmd.value > 1` always going to evaluate to `true` unless `cmd.value` is `0`? Would `0 < cmd.value < 33` make more sense?

Comment (dkirker) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: Or I assume this was meant to be `cmd.value < 33 && cmd.value > 0`.

Comment (greens) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: you are correct

File: devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy [hunk 2]
```diff
@@ -111,7 +111,7 @@ def fanEvents(physicalgraph.zwave.Command cmd) {
 	def result = [createEvent(name: "switch", value: value)]
 	result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value)
 	def fan_level = Math.round(cmd.value/33)
-	if (cmd.value < 33 * cmd.value > 1) fan_level = 1 //sometimes we get "1" when the device is on low
+	if (cmd.value < 33 && cmd.value > 1) fan_level = 1 //sometimes we get "1" when the device is on low
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: You'll either want `cmd.value > 0` or `cmd.value >= 1` if we get sent `1` from the device.

Comment (greens) [reply devicetypes/smartthings/zwave-fan-controller.src/zwave-fan-controller.groovy]: Very true!

**Comment by dkirker** [comment]
The Math.ceil route looks good. If it tests ok then +1.

**Review by dkirker — DISMISSED** [review]

**Review by dkirker — APPROVED** [review]

**Pull request merged**